frontend/app/routes/_public+/apply+/$id+/demographics-part1.tsx

```diff
@@ -27,21 +33,121 @@ export const meta: MetaFunction<typeof loader> = mergeMeta((args) => {
 export async function loader({ request, params }: LoaderFunctionArgs) {
   const bornTypes = await getLookupService().getAllBornTypes();
   const disabilityTypes = await getLookupService().getAllDisabilityTypes();
+  const { OTHER_EQUITY_TYPE_ID } = getEnv();
+  const equityTypes = await getLookupService().getAllEquityTypes();
   const applyFlow = getApplyFlow();
   const { id, state } = await applyFlow.loadState({ request, params });
-  return json({ id, state, bornTypes, disabilityTypes });
+
+  const otherEquityCode = equityTypes.find((equityType) => equityType.id === OTHER_EQUITY_TYPE_ID);
+  if (!otherEquityCode) {
+    throw new Response(`Unexpected 'Other' equity type: ${OTHER_EQUITY_TYPE_ID}`, { status: 500 });
+  }
+
+  return json({ id, bornTypes, disabilityTypes, otherEquityCode, equityTypes, state: state.demographicsPart1 });
 }
 
-export async function action({ request }: ActionFunctionArgs) {
-  return json('');
+export async function action({ request, params }: ActionFunctionArgs) {
+  const applyFlow = getApplyFlow();
+  const { id } = await applyFlow.loadState({ request, params });
+
+  const formSchema = z.object({
+    bornType: z.string({ required_error: 'empty-radio' }),
+    equity: z.string({ required_error: 'empty-radio' }),
+    otherEquity: z.string().min(1, { message: 'empty-field' }).optional(),
+    disabilityType: z.string({ required_error: 'empty-radio' }),
+  });
+
+  const formData = Object.fromEntries(await request.formData());
+  const parsedDataResult = formSchema.safeParse(formData);
+
+  if (!parsedDataResult.success) {
+    return json({
+      errors: parsedDataResult.error.format(),
+      formData: formData as Partial<z.infer<typeof formSchema>>,
+    });
+  }
+
+  const sessionResponseInit = await applyFlow.saveState({
+    request,
+    params,
+    state: { demographicsPart1: parsedDataResult.data },
+  });
+
+  return redirect(`/apply/${id}/review-information`, sessionResponseInit);
 }
 
 export default function DemographicsPart1() {
-  const { bornTypes, disabilityTypes } = useLoaderData<typeof loader>();
+  const { bornTypes, disabilityTypes, otherEquityCode, equityTypes, state } = useLoaderData<typeof loader>();
   const { i18n, t } = useTranslation(handle.i18nNamespaces);
+  const [otherEquityChecked, setOtherEquityChecked] = useState(state?.equity === otherEquityCode.id);
+  const actionData = useActionData<typeof action>();
+  const errorSummaryId = 'error-summary';
+
+  function otherEquityHandler() {
+    setOtherEquityChecked(true);
+  }
+
+  function nonOtherEquityHandler() {
+    setOtherEquityChecked(false);
+  }
+
+  /**
+   * Gets an error message based on the provided internationalization (i18n) key.
+   *
+   * @param errorI18nKey - The i18n key for the error message.
+   * @returns The corresponding error message, or undefined if no key is provided.
+   */
+  function getErrorMessage(errorI18nKey?: string): string | undefined {
```

Comment:
**Nitpick:** typescript is usually pretty good at inferring the return type of a function. You shouldn't need to specify it unless the return type is ambiguous and you need to pass the function as a parameter to another function.

You don't have to change it here, just leaving a comment so you know for your future PRs.